esp-hal/src/peripherals.rs

```diff
@@ -119,10 +119,29 @@ macro_rules! create_peripheral {
     };
 }
 
+macro_rules! io_type {
+    (Input, $peri:ident) => {
+        impl $crate::gpio::InputPin for $peri<'_> {
+            #[doc(hidden)]
+            #[inline(always)]
```

Comment:
Another `inline(always)` that probably shouldn't be here? Maybe not, given that this is a impl, not not a trait definition, but I thought I'd ask :D.